### ISSUE TITLE
Feature suggestion: Download sql admin result

### ISSUE THREAD

**Issue by Ripipip** [Starcounter/Starcounter#3474]
I don't know the best place to suggest features but if it would be possible to download the result you get in the sql admin panel it would be extremely valueable for us. It is already formatted similar to an excel sheet and if we could just export it to excel we would save a lot of time and effort.

**Comment by miyconst** [comment]
@Ripipip thank you for the request!

I have changed the issue title to be more understandable. The feature itself should not be difficult to implement if you are fine with `.xlsx` document.

@andwah how much time would this take? I can help you with MS-Excel document generation.

**Comment by andwah** [comment]
This as to be done:

-  Add Export button in gui. (easy&fast to develop)
-  Add new handler for the file generation that will return the file. (easy&fast to develop)
-  Prepare the query result.

Today we convert the query result to JSON and resturn it to the caller.

**Fastes developing time (1day) **: 
      Going to go from Query Result -> JSON -> .xlsx

**Best performence & implementation**
     Going from query result direct to .xlsx
But it will take about 3 days or so to develop and test,

**Comment by Ripipip** [comment]
People over here got seriously excited when I accidently mentioned the possibility of being able to download query result...  So whatever it is worth I would say that the faster it can be done the happier we would be :)

**Comment by andwah** [comment]
Maybe we should implement the export function in a public 'helper' library. then developers may also use it within their apps. The Admin will also use the same principe as it is also an app.

**Comment by miyconst** [comment]
>Going to go from Query Result -> JSON -> .xlsx

This one is the way to go I think. 

@andwah let's do this after the Warehouse demo.

**Comment by andwah** [comment]
> Going to go from Query Result -> JSON -> .xlsx
> This one is the way to go I think.

I dont agree, i think we should aim for performence and a good implmentation..even if it takes 2 more extra days. But let @dan31 deside.

**Comment by miyconst** [comment]
This feature will be rarely used and it is always good to have a json-to-excel component which can be used later in other places.

Anyway, let's decide on it after the Warehouse demo.

**Comment by k-rus** [comment]
> Maybe we should implement the export function in a public 'helper' library.

I don't like the idea of having the functionality in the helper library. It is not something crucial for apps. Furthermore, the maintenance cost is too high for us. Converting JSON does not require much maintenance and some external tools can do it, I guess.

**Comment by miyconst** [comment]
I agree with @k-rus.

**Comment by andwah** [comment]
I added the ability to export to an excel sheet in the DatabaseEditor app (v1.0.7) as an Test, play around and see what you think.  The code can then easy be moved into the Administrator.

**Note**: to Export you need to **RIGHT CLICK** and select **SAVE LINK AS...** on the `Export `link


![image](https://cloud.githubusercontent.com/assets/2656996/14330540/51cd4562-fc41-11e5-9854-22a59a7a6b3b.png)


The DatabaseEditor app can be found and installed from the `App Warehouse` tab in the Administrator.
